File: frontend/src/utils/ownerName.ts
export function parseOwnerName(raw: string): {
  first_name: string
  last_name: string
} {
  const s = (raw ?? '').trim()
  if (!s) return { first_name: '', last_name: '' }
  const commaIdx = s.indexOf(',')
  if (commaIdx >= 0) {
    return {
      first_name: s.slice(0, commaIdx).trim(),
      last_name: s.slice(commaIdx + 1).trim(),
    }
  }
  const match = s.match(/^(\S+)\s+(.+)$/)
  if (!match) return { first_name: s, last_name: '' }
  return { first_name: match[1], last_name: match[2].trim() }
}

File: backend/app/owner_name.py
def parse_owner_name(raw: str) -> tuple[str, str]:
    """Parse a freeform name into (first_name, last_name).

    Comma takes priority — first name is in front of the comma:
        "Doe, John" -> ("Doe", "John")
    Otherwise split on the first whitespace run:
        "John Smith" -> ("John", "Smith")
        "Cher"       -> ("Cher", "")
    Empty / whitespace-only input returns ("", "").
    """
    if raw is None:
        return ("", "")
    s = raw.strip()
    if not s:
        return ("", "")
    if "," in s:
        first, _, last = s.partition(",")
        return (first.strip(), last.strip())
    parts = s.split(None, 1)
    if len(parts) == 1:
        return (parts[0], "")
    return (parts[0], parts[1].strip())
